### PR TITLE
python39Packages.oci: 2.56.0 -> 2.60.0, oci-cli: 3.5.0 -> 3.6.0, lexicon: 3.9.0 -> 3.9.4

### PR DESCRIPTION
**Pull request by WolfangAukang**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by WolfangAukang** [comment]
oci-cli needs to be updated to version 3.5.3 ([check pyproject.toml](https://github.com/oracle/oci-cli/commit/7adcdb983c5d1ec0149b6854bf21f93c73b8c526)).

Tests failed for dnspython. Upstream is at 2.2.0, while nixpkgs still has 1.16.0.

**Comment by WolfangAukang** [comment]
Result of `nixpkgs-review pr 163010` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>lexicon</li>
    <li>oci-cli</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.oci</li>
    <li>python39Packages.oci</li>
  </ul>
</details>

**Comment by WolfangAukang** [comment]
Result of `nixpkgs-review pr 163010` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>lexicon</li>
    <li>oci-cli</li>
    <li>python310Packages.oci</li>
    <li>python39Packages.oci</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -3406,7 +3406,7 @@ with pkgs;
 
   lepton-eda = callPackage ../applications/science/electronics/lepton-eda { };
 
-  lexicon = callPackage ../tools/admin/lexicon { };
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:3409]: We try to avoid mixing of package scopes, it should be using the `pkgs.callPackage`

Comment (WolfangAukang) [reply pkgs/top-level/all-packages.nix:3409]: But building it this way wouldn't recognize buildPythonApplication:

```
·> nix-build -A lexicon
error: anonymous function at /home/syane/Projektujo/Aliaj/nixpkgs/pkgs/tools/admin/lexicon/default.nix:1:1 called without required argument 'buildPythonApplication'

       at /home/syane/Projektujo/Aliaj/nixpkgs/lib/customisation.nix:69:16:

           68|     let
           69|       result = f origArgs;
             |                ^
           70|
```

File: pkgs/tools/admin/lexicon/default.nix [hunk 2]
```diff
@@ -1,44 +1,38 @@
...
-
-      localzone = super.localzone.overridePythonAttrs(oldAttrs: rec {
-        meta = oldAttrs.meta // { broken = false; };
-      });
-    };
-  };
-in
-  with py.pkgs;
-
 buildPythonApplication rec {
```
Comment (fabaff) [inline pkgs/tools/admin/lexicon/default.nix:26]:
```suggestion
python3.pkgs.buildPythonApplication rec {
```

File: pkgs/tools/admin/lexicon/default.nix [hunk 3]
```diff
@@ -1,44 +1,38 @@
 { lib
-, python3
+, buildPythonApplication
```
Comment (fabaff) [inline pkgs/tools/admin/lexicon/default.nix:2]: Change not needed -> https://github.com/NixOS/nixpkgs/pull/163010/files#r824255580

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/top-level/all-packages.nix:3409]: With https://github.com/NixOS/nixpkgs/pull/163010/files#r824255580 this change is not needed.

File: pkgs/tools/admin/lexicon/default.nix (hunk 2 above)
Comment (WolfangAukang) [reply pkgs/tools/admin/lexicon/default.nix:8]: Is there no issues with using with python3.pkgs instead?

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (WolfangAukang) [reply pkgs/top-level/all-packages.nix]: Done

File: pkgs/tools/admin/lexicon/default.nix (hunk 3 above)
Comment (WolfangAukang) [reply pkgs/tools/admin/lexicon/default.nix]: Done

**Comment by WolfangAukang** [comment]
Result of `nixpkgs-review pr 163010` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>lexicon</li>
    <li>oci-cli</li>
    <li>python310Packages.oci</li>
    <li>python39Packages.oci</li>
  </ul>
</details>

**Pull request merged**